### PR TITLE
My cases for Judicial and Legla ops

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-4395

### Change description ###

Introduce 2 column configurations. Although the net outcome of UI should be the same. This is preparation for other JIRAs EUI-3614 and EUI-3640

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by kuda1992 — APPROVED** [review]
Looks good

File: src/work-allocation-2/components/constants/config.constants.ts [hunk 1]
```diff
@@ -118,17 +118,22 @@ const ASSIGNEE: FieldConfig = {
 /**
  * The views.
  */
-const AVAILABLE_TASKS: FieldConfig[] = [
+const AVAILABLE_TASKS_FOR_JUDICIAL: FieldConfig[] = [
+  CASE_NAME_AS_TEXT, CASE_CATEGORY, LOCATION, DERIVED_ICON, TASK_NAME_AS_TEXT, DUE_DATE
+];
+const AVAILABLE_TASKS_FOR_LEGAL_OPS: FieldConfig[] = [
   CASE_NAME_AS_TEXT, CASE_CATEGORY, LOCATION, DERIVED_ICON, TASK_NAME_AS_TEXT, DUE_DATE
```
Comment (rayliangatos) [inline src/work-allocation-2/components/constants/config.constants.ts:125]: looks like these two fields configs have same setting, why they are splited to different views?

File: src/work-allocation-2/components/constants/config.constants.ts [hunk 2]
```diff
@@ -118,17 +118,22 @@ const ASSIGNEE: FieldConfig = {
...
 const MY_TASKS: FieldConfig[] = [
   CASE_REFERENCE_AS_LINK, CASE_NAME_AS_TEXT, CASE_CATEGORY, LOCATION, DERIVED_ICON, TASK_NAME_AS_TEXT, DUE_DATE
 ];
-const MY_WORK_TASKS: FieldConfig[] = [
+const MY_WORK_TASKS_FOR_JUDICIAL: FieldConfig[] = [
+  CASE_NAME_AS_LINK, CASE_CATEGORY, LOCATION, DERIVED_ICON, TASK_NAME_AS_LINK, DUE_DATE
+];
+const MY_WORK_TASKS_FOR_LEGAL_OPS: FieldConfig[] = [
   CASE_NAME_AS_LINK, CASE_CATEGORY, LOCATION, DERIVED_ICON, TASK_NAME_AS_LINK, DUE_DATE
 ];
```
Comment (rayliangatos) [inline src/work-allocation-2/components/constants/config.constants.ts:138]: looks like these two fields configs have same setting, why they are splited to different views?

File: src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts [hunk 3]
```diff
@@ -297,4 +299,13 @@ export class TaskListWrapperComponent implements OnDestroy, OnInit {
...
 
+  public isCurrentUserJudicial(): boolean {
+    const userInfoStr = this.sessionStorageService.getItem(this.userDetailsKey);
+    if (userInfoStr) {
+      const userInfo: UserInfo = JSON.parse(userInfoStr);
+      const isJudge = userInfo.roles.some(role => ListConstants.JUDGE_ROLES.includes(role));
+      return isJudge;
+    }
+    return false
+  }
```
Comment (rayliangatos) [inline src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts:310]: could this use the existing AppUtils.isLegalOpsOrJudicial(roles).

File: src/work-allocation-2/components/constants/config.constants.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/work-allocation-2/components/constants/config.constants.ts:125]: This is a preparation for the other JIRAs. Please see descrption

File: src/work-allocation-2/components/constants/config.constants.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/work-allocation-2/components/constants/config.constants.ts:138]: This is a preparation for the other JIRAs. Please see description

File: src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts (hunk 3 above)
Comment (udaydenduluri33) [reply src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts:310]: No, that seems to be different one

Comment (udaydenduluri33) [reply src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts]: Done, changed

**Review by rayliangatos — APPROVED** [review]

**Pull request merged**